assets/js/components/audit-opinions.js

```diff
@@ -20,14 +20,14 @@ class ReportCard {
     this.$element.find(".audit-outcome__year").text(formatFinancialYear(report.date));
     this.$element.find(".audit-outcome__heading").text(report.result);
 
-    if (report.report_url.endsWith('.pdf')) {
+    if (report.report_url === null) {
+      this.$element.find(".audit-outcome__download").text("No report available");
+    }
+    else if (report.report_url.endsWith('.pdf')) {
       report.report_url = report.report_url.substring(0, report.report_url.lastIndexOf("/"));
     }
 
     this.$element.find(".audit-outcome__download").attr("href", report.report_url);
-    if (report.report_url === null) {
-      this.$element.find(".audit-outcome__download").text("No report available");
-    }
```

Comment:
I see now this is modifying someone else's data. Similarly to https://github.com/OpenUpSA/municipal-data/pull/578/files#r853249892, that could have unexpected consequences. Better style would be to create a local variable that is report.report_url if it doesn't end with .pdf, otherwise we strip the pdf and assign that. And then use the local variable for the PDF URL.

Reply:
Added changes to address this